movement/watch_faces/complication/randonaut_face.c

```diff
@@ -0,0 +1,411 @@
+                    __0x2_(0x2),__0x2_(23));___0xf322(/*____0x2222*/
+                    /*0x2*/__0x2_(002),__0x2_(12+7+11));/*___00x22*/
+                    if(/*_0x22*/c<70){_0x22(__0x2_(2),/*____0x2222*/
+                    __0x2_(12+7+11));}if(c<60){_0x22(/*_______0x22*/
+                    /*_0x22*/__0x2_(002),__0x2_(0xD68-/*_0x2222222*/
+                    0xD4A));}if(c<50){_0x22(__0x2_(0x2),/*____0x22*/
+                    __0x2_(14+9));}delay_ms(__0x2_(c)+20);if/*_0x2*/
+                    (c<30){watch_display_string(" ",__0x2_(/*_2**2*/
+                    10));}_0x22(__0x2_(02),__0x2_(3432-3409)/*0x22*/
+                    );_0x22(__0x2_(002),__0x2_(51-28));/*_____0x22*/
+                    /**/  _0x22(__0x2_(0x2),__0x2_(23));if(c<20)////
+                    /*_*/{_0x22(__0x2_(02),__0x2_(51-28));/*__0x22*/
+                    /*_0x22*/_0x22(__0x2_(2),__0x2_(14+9));/*_0x22*/
+                    /*_0x22*/_0x22(__0x2_(0x2),__0x2_(0xD68-0xD4A));
+                    /*_0x22*/_0x22(__0x2_(0x2),__0x2_(3432-3409));//
+                    /*_0x22*/_0x22(__0x2_(002),__0x2_(12+7+11));////
+                    /*_0x22**_0x22*/_0x22(__0x2_(2),__0x2_(51-28));}
+                }
+            else
+                for ( uint8_t c = 30; c > 0; c--) {
+                    watch_display_string("1", _get_pseudo_entropy(10));
+                    watch_display_string("0", _get_pseudo_entropy(10));
+                    watch_display_string("11", _get_pseudo_entropy(10));
+                    watch_display_string("00", _get_pseudo_entropy(10));
+                    delay_ms(50);
+                    watch_display_string(" ", _get_pseudo_entropy(10));
+                    watch_display_string(" ", _get_pseudo_entropy(10));
+                    watch_display_string(" ", _get_pseudo_entropy(10));
+                    watch_display_string(" ", _get_pseudo_entropy(10));
+                }
+            _generate_blindspot(state);
+            watch_clear_display();
+            state->face.mode = 2; // point
+            state->face.location_format = 1; // distance
+            watch_display_string("RA   Found", 0);
+            delay_ms(500);
+            sprintf(buf, "RA   Found");
+            break;
+        case 2: //point
+            switch ( state->face.location_format ) {
+                case 0:
+                    sprintf(buf, "RA   Point");
+                    break;
+                case 1: // distance to point
+                    watch_clear_display();
+                    sprintf(buf, "DI m %d", state->point.distance );
+                    break;
+                case 2: // bearing relative to point
+                    watch_clear_display();
+                    sprintf(buf, "BE # %d", state->point.bearing );
+                    break;
+                case 3: // latitude DD._____
+                    sprintf(state->scratchpad, "%07d", abs(state->point.latitude));
```

Comment:
My compiler fires a warning on this line about taking the absolute value of an unsigned int, which is weird because your data structure defines it as a signed int? Can you check on whether you get the same warning? Specifically: 

```
../watch_faces/complication/randonaut_face.c:285:56: warning: taking the absolute value of unsigned type 'unsigned int:26' has no effect [-Wabsolute-value]
```

Reply:
Oh yes I got the same warning and was confused about it. I looked into it and it seems to have to do with the signed int being bitmasked from 32 bit to 26 so if I were to assign a real int32 to that struct element I could lose the bit that is flags the sign. But since I am not doing that it should be fine I guess?

```C
typedef struct {
    int32_t latitude : 26;
    int32_t longitude : 26;
    uint16_t distance : 14;
    uint16_t bearing : 9;
} randonaut_coordinate_t;
```